### PR TITLE
feat(panels): adds panel component

### LINKED ISSUE #138: Add Panel Component
## 🚀 Feature Proposal

Add a Panel component to the component library.

The Panel component should support: 
* An optional header
* A required body
* An optional footer
* Ability to be collapsible 

## Motivation

Grouping of like items. 

## Example

Please provide an example for how this feature would be used.
image

### PR DESCRIPTION
**Pull request by MatthewDorner**
Fixes #138.

**Changes proposed in this pull request:**
- An optional header
- A required body
- An optional footer
- Ability to be collapsible

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -34,14 +36,16 @@ function getFontAwesomeIcon(icon: IconType): string {
 interface Props {
   /** The type of icon to display */
   icon: IconType
+  /** Function to execute when user clicks on icon */
+  onClick?: (event: React.MouseEvent<any>) => void
```
Comment (jackcmeyer) [inline src/components/Icon/Icon.tsx]: There should be a test to make sure that the onClick gets passed to the FontAwesomeIcon properly and that the `onClick` prop gets fired when the fontawesome icon is clicked.

File: src/components/Panel/Panel.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  title?: string
+  /** The footer for the panel */
+  footer?: string
+  /** Determines if the panel can be collapsible */
+  collapsible?: boolean
+  /** Determines if the panel should be collapsed */
+  collapsed?: boolean
+}
+
+const Panel = (props: Props) => {
```
Comment (jackcmeyer) [inline src/components/Panel/Panel.tsx]: There should be some tests for the Panel component prop.

**Comment by jackcmeyer** [comment]
Welcome back, @MatthewDorner 🎉😊

File: src/components/Panel/Panel.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react'
+import { Card, Collapse } from 'react-bootstrap'
+import { Icon } from '../Icon'
+
+interface Props {
+  /** Defines the color of the panel */
+  color?: 'primary' | 'secondary' | 'success' | 'warning' | 'danger' | 'info' | 'light' | 'dark'
```
Comment (jackcmeyer) [inline src/components/Panel/Panel.tsx]: this same set of colors are defined in the `ButtonColor` interface in `components/Button/interfaces/ts`. We could probably extract this to a common interface.

Comment (MatthewDorner) [reply src/components/Panel/Panel.tsx]: Where should the common interface reside? a new root folder?

Comment (jackcmeyer) [reply src/components/Panel/Panel.tsx]: Seems fine to me.

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (MatthewDorner) [reply src/components/Icon/Icon.tsx]: added

File: src/components/Panel/Panel.tsx (hunk 2 above)
Comment (MatthewDorner) [reply src/components/Panel/Panel.tsx]: added

File: src/components/Panel/Panel.tsx (hunk 3 above)
Comment (MatthewDorner) [reply src/components/Panel/Panel.tsx]: done

**Comment by fox1t** [comment]
LGTM

**Pull request merged**